### PR TITLE
FEATURE added pickFlowersFromPots rule.

### PR DESCRIPTION
**Pull request by Serzbiedry**
Injects into FlowerPotBlock.onUse at tail, essentially making another case for interaction.

### CODE REVIEW

File: src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                     CallbackInfoReturnable<ActionResult> cir) {
+    Block potBlock = state.getBlock();
+    Block content;
+    if (!CarpetAddonsNotFoundSettings.pickFlowersFromPots || !(potBlock instanceof FlowerPotBlock) ||
+        (content = ((FlowerPotBlock) potBlock).getContent()) == Blocks.AIR) {
+      cir.cancel();
+      cir.setReturnValue(ActionResult.CONSUME);
+      return;
+    }
+    world.setBlockState(pos, Blocks.FLOWER_POT.getDefaultState(), Block.NOTIFY_ALL);
```
Comment (Gilly7CE) [inline src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java:44]: I think we also need the following line: `world.emitGameEvent((Entity)player, GameEvent.BLOCK_CHANGE, pos);`

I'm suggesting this as that is what the `FlowerPotBlock.onUse` does before it sets the `ActionResult` to success.

File: src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                     PlayerEntity player,
+                     Hand hand,
+                     BlockHitResult hit,
+                     CallbackInfoReturnable<ActionResult> cir) {
+    Block potBlock = state.getBlock();
+    Block content;
+    if (!CarpetAddonsNotFoundSettings.pickFlowersFromPots || !(potBlock instanceof FlowerPotBlock) ||
+        (content = ((FlowerPotBlock) potBlock).getContent()) == Blocks.AIR) {
+      cir.cancel();
+      cir.setReturnValue(ActionResult.CONSUME);
```
Comment (Gilly7CE) [inline src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java:41]: This isn't needed. I've tested this locally too to make sure and I had no issues.

File: src/main/resources/assets/carpet-addons-not-found/lang/en_us.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
   "carpet.rule.netheritePickaxeInstantMineEndStone.desc": "A netherite pickaxe with efficiency V combined with the haste II status effect will instant mine end stone type blocks.",
   "carpet.rule.netheritePickaxeInstantMineNetherBricks.desc": "A netherite pickaxe with efficiency V combined with the haste II status effect will instant mine nether brick type blocks.",
   "carpet.rule.phantomsObeyHostileMobCap.desc": "Phantoms will no longer spawn if the hostile mobcap is full. This is per player.",
+  "carpet.rule.pickFlowersFromPots.desc": "Right-clicking on a flowerpot with a flower in it will put the flower into your invantory",
```
Comment (Gilly7CE) [inline src/main/resources/assets/carpet-addons-not-found/lang/en_us.json:21]: Description should be "Right-clicking on a flowerpot containing a flower will always put the flower into your inventory."

File: README.md [hunk 4]
```diff
@@ -204,6 +204,15 @@ Phantoms will no longer spawn if the hostile mobcap is full. This is per player.
 - Required options: `true`, `false`
 - Categories: `FEATURE`, `CARPET-ADDONS-NOT-FOUND`
 
+### pickFlowersFromPots
+
+Right-clicking on a flowerpot with a flower in it will put the flower into your invantory.
```
Comment (Gilly7CE) [inline README.md:209]: Description should be "Right-clicking on a flowerpot containing a flower will always put the flower into your inventory."

File: src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  private void onUse(BlockState state,
+                     World world,
+                     BlockPos pos,
+                     PlayerEntity player,
+                     Hand hand,
+                     BlockHitResult hit,
+                     CallbackInfoReturnable<ActionResult> cir) {
+    Block potBlock = state.getBlock();
+    Block content;
+    if (!CarpetAddonsNotFoundSettings.pickFlowersFromPots || !(potBlock instanceof FlowerPotBlock) ||
```
Comment (Basicprogrammer10) [inline src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java:38]: I think you can just do:
```java
... !(potBlock instanceof FlowerPotBlock content) || content.getContent() == Blocks.AIR ...
```

Which will set `content` to `(FlowerPotBlock) potBlock` if `potBlock` is an instance of `FlowerPotBlock`.

File: README.md [hunk 6]
```diff
@@ -204,6 +204,15 @@ Phantoms will no longer spawn if the hostile mobcap is full. This is per player.
 - Required options: `true`, `false`
 - Categories: `FEATURE`, `CARPET-ADDONS-NOT-FOUND`
 
+### pickFlowersFromPots
```
Comment (Gilly7CE) [inline README.md:207]: The rule name should probably be `alwaysPickFlowersFromPots` to distinguish it from the vanilla behaviour with an empty hand.

File: src/main/java/carpetaddonsnotfound/CarpetAddonsNotFoundSettings.java [hunk 7]
```diff
@@ -63,6 +63,9 @@ public class CarpetAddonsNotFoundSettings {
   @Rule(categories = { FEATURE, CARPET_ADDONS_NOT_FOUND })
   public static boolean phantomsObeyHostileMobCap = false;
 
+  @Rule(categories = { FEATURE, CARPET_ADDONS_NOT_FOUND })
+  public static boolean pickFlowersFromPots = false;
```
Comment (Gilly7CE) [inline src/main/java/carpetaddonsnotfound/CarpetAddonsNotFoundSettings.java:67]: See previous comment about the rule name.

File: src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java (hunk 2 above)
Comment (Basicprogrammer10) [inline src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java:41]: Calling `setReturnValue` implicitly calls `cancel`. You don't need to also call cancel yourself.

**Review by Gilly7CE — CHANGES_REQUESTED** [review]
Couple of things to address, otherwise good stuff!

**Review by Basicprogrammer10 — CHANGES_REQUESTED** [review]

File: src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java (hunk 2 above)
Comment (Gilly7CE) [reply src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java:41]: See my comment as none of this setting return value is needed.

**Comment by Gilly7CE** [comment]
Closing now that #139 has been opened

**Pull request closed without merging**